### PR TITLE
[Task][56164] Thêm tính năng cho SDK [Deploy] Deploy 24/11/2022

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task 56164: Thêm tính năng cho SDK](https://edu-redmine.sun-asterisk.vn/issues/56164)

## WHAT

1. Get balance by address
2. Send ICX
3. Get transaction result

## Evidence (Screenshot or Video)

- N/A

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/Zts4P8 |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Message | Sun* commit message | Template: [TASK/FREATURE/BUG/SUPPORT][RedmineID] + message |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: Whollet/Whollet/Source/Data/Repositories/WalletRepositoryImpl.swift [hunk 1]
```diff
@@ -13,4 +13,52 @@ final class WalletRepositoryImpl: WalletRepository {
...
+            return "\(balance)"
+        case .failure(_):
+            return "0"
+        }
+    }
+    
+    func sendICX(to: String, amount: Double, wallet: Wallet) -> String? {
+        let coinTransfer = Transaction()
+            .from(wallet.address)
+            .to(to)
```
Comment (ghost) [inline Whollet/Whollet/Source/Data/Repositories/WalletRepositoryImpl.swift]: có tên nào hắn oke hơn k e

File: Whollet/Whollet/Source/Data/Repositories/WalletRepositoryImpl.swift [hunk 2]
```diff
@@ -13,4 +13,52 @@ final class WalletRepositoryImpl: WalletRepository {
...
+            let request = iconService.sendTransaction(signedTransaction: signed)
+            let response = request.execute()
+            switch response {
+            case .success(let result):
+                return result
+            case .failure(_):
+                return nil
+            }
+        } catch {
+            return nil
```
Comment (ghost) [inline Whollet/Whollet/Source/Data/Repositories/WalletRepositoryImpl.swift]: sao catch lỗi hay response failed lại trả về nil e ?

File: Whollet/Whollet/Source/Domain/UseCases/GetICXBalanceUseCase.swift [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import Foundation
+
+final class GetICXBalanceUseCase {
+    private let repository = WalletRepositoryImpl()
+    func get(address: String) -> String {
+        return repository.getIcxBalance(address: address)
+    }
```
Comment (ghost) [inline Whollet/Whollet/Source/Domain/UseCases/GetICXBalanceUseCase.swift]:
```suggestion
    private let repository = WalletRepositoryImpl()
    
    func get(address: String) -> String {
        return repository.getIcxBalance(address: address)
    }
```

File: Whollet/Whollet/Source/Domain/UseCases/GetTransactionResultUseCase.swift [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import Foundation
+import ICONKit
+
+final class GetTransactionResultUseCase {
+    private let repository = WalletRepositoryImpl()
+    func get(txHash: String) -> Response.TransactionByHashResult? {
+        return repository.getTransaction(txHash: txHash)
+    }
```
Comment (ghost) [inline Whollet/Whollet/Source/Domain/UseCases/GetTransactionResultUseCase.swift]: as above

File: Whollet/Whollet/Source/Domain/UseCases/SendICXUseCase.swift [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import Foundation
+import ICONKit
+
+final class SendICXUseCase {
+    private let repository = WalletRepositoryImpl()
+    func send(to: String, amount: Double, wallet: Wallet) -> String? {
+        return repository.sendICX(to: to, amount: amount, wallet: wallet)
+    }
```
Comment (ghost) [inline Whollet/Whollet/Source/Domain/UseCases/SendICXUseCase.swift]: as above

File: Whollet/Whollet/Source/Data/Repositories/WalletRepositoryImpl.swift (hunk 1 above)
Comment (ghost) [reply Whollet/Whollet/Source/Data/Repositories/WalletRepositoryImpl.swift]: sendTransaction :v em mới đổi xong

**Review by tunghs-1253 — APPROVED** [review]

**Pull request merged**